account-integrations/safe/test/hardhat/integration/SafeECDSAPlugin.test.ts

```diff
@@ -51,8 +49,11 @@ describe("SafeECDSAPlugin", () => {
     }
 
     return {
-      factory: factory.attach(SAFE_FACTORY_ADDRESS).connect(userWallet),
-      singleton: singleton.attach(SINGLETON_ADDRESS).connect(provider),
+      factory: SafeProxyFactory__factory.connect(
+        SAFE_FACTORY_ADDRESS,
+        userWallet,
+      ),
+      singleton: Safe__factory.connect(SINGLETON_ADDRESS, provider),
```

Comment:
This results in the same JS entities, but doing it this way results in better type information. I recommend avoiding `getContractFactory` in general, because the way it works is less robust than the typechain alternatives.